### PR TITLE
FLY-2588: add Codecov coverage reporting

### PR DESCRIPTION
**Pull request by Freshenext**
## What

Adds Codecov coverage reporting to this repo. A `codecov.yml` sets the project and patch checks, the CI build job now runs `forge coverage` into an LCOV report and uploads it, and the README shows a coverage badge.

## Why

Today a PR here gives reviewers no idea whether the changed lines are tested. The commented-out Coveralls block in `ci.yml` was the last attempt and it never ran. `liquidity-provider-server` already publishes a coverage delta on every PR, so this brings LBC to the same place with the same thresholds.

## Type of Change

- [ ] Bug fix (non-breaking change)
- [ ] New feature (non-breaking change)
- [ ] Breaking change
- [ ] Refactor (no intended behavior change)
- [ ] Security hardening
- [ ] Tests only
- [x] Build/CI/Tooling
- [ ] Documentation

## Affected Areas

- [ ] PegIn flow (`registerPegIn`, `callForUser`, quote validation)
- [ ] PegOut flow (`depositPegout`, `refundPegOut`, `refundUserPegOut`)
- [ ] Liquidity Provider lifecycle (register/update/status/resign/collateral/withdraw)
- [ ] Pause/operational controls
- [ ] Quote hashing/signature logic
- [ ] Bitcoin tx / PMT / merkle validation
- [x] Deployment or upgrade scripts / Makefile targets
- [x] CI workflows (`ci`, `fuzz`, `formal`, `slither`, `codeql`)
- [x] Docs

## Risk & Security Review

- [ ] Access control and authorization paths reviewed
- [ ] Reentrancy and external call paths reviewed
- [ ] Storage layout / upgrade safety reviewed (if upgradeable code changed)
- [ ] Time/block/confirmation assumptions reviewed
- [ ] BTC tx output/index assumptions reviewed (if pegout validation touched)
- [x] No secrets or sensitive values added

No Solidity changed. The upload step reads `secrets.CODECOV_TOKEN`, which does not exist yet, and the action is pinned by SHA like every other action in this workflow.

## Test Plan

List what you ran locally and the outcome:

- [x] `npm run lint:sol`
- [x] `npm run lint:ts`
- [x] `npm run compile`
- [x] `npm test`
- [ ] `npm run test:fuzz` (if logic/state transitions changed)
- [ ] `npm run test:invariant` (if invariant-sensitive logic changed)
- [ ] `npm run test:formal` (if formal-verification-sensitive logic changed)
- [ ] `npm run test:integration` (if integration paths changed)
- [ ] `npm run test:e2e` (if deployment/ownership flows changed)

Additional notes on test coverage, edge cases, and any tests intentionally skipped:

- Also ran the new `npm run coverage:lcov` on Foundry 1.6.0. 520 tests passed and `coverage/lcov.info` was written, covering 111 source files, 52 of them under `src/`.
- `codecov.yml` was checked against the Codecov validator API (`curl --data-binary @codecov.yml https://codecov.io/validate`), which returned `Valid!`.
- Both YAML files parse with `js-yaml`.
- Fuzz, invariant, formal, integration and e2e suites were not run. This PR touches no contract logic, and the coverage target deliberately excludes those suites to keep the CI job at unit-test runtime.
- The Codecov upload itself could not be exercised locally. It needs the token and a Codecov-side repo (see Reviewer Notes).

## Deployment & Ops Impact

- [x] No deployment impact
- [ ] Requires deployment
- [ ] Requires upgrade
- [ ] Env/config changes required (`.env`, network RPCs, verification, etc.)
- [ ] Runbook/update instructions added to PR description

One GitHub Actions secret is required (below). Nothing to deploy on chain.

## Related Issues

- Jira: [FLY-2588](https://rsklabs.atlassian.net/browse/FLY-2588)
- GitHub Issue: n/a

## Reviewer Notes

The upload step will not work until Codecov onboarding and a repo-scoped `CODECOV_TOKEN` Actions secret are provisioned for `rsksmart/liquidity-bridge-contract`. That is tracked as FLY-2600, currently unassigned. Until it lands, the "Upload coverage to Codecov" step fails, because `fail_ci_if_error: true` matches the LPS setup. If you would rather not have a red CI in the gap, either merge this after FLY-2600 or say so and I will flip the flag for now.

Three things worth a look:

1. The badge points at `branch/v2.6.0`, since that is the integration branch, not `master`.
2. `make coverage-lcov` reuses the `test-unit` path exclusions (`fuzz`, `invariant`, `differential`, `formal`). Including them would slow the CI job considerably and would credit coverage to randomized runs.
3. The `ignore:` list in `codecov.yml` drops `test/`, `script/`, `src/test/` and `src/test-contracts/`. `src/legacy/` is kept, since it is deployed code.

Raising the coverage number and making the Codecov check required for merge are both out of scope here.

## Screenshots / Logs (if applicable)

Local run of the new target:

```
$ npm run coverage:lcov
Running tests with coverage (LCOV report)...
Ran 63 test suites in 25.31s: 520 tests passed, 0 failed, 0 skipped (520 total tests)
Wrote LCOV report.

$ head -2 coverage/lcov.info
TN:
SF:script/ComputeStorageSlot.s.sol
```

Codecov config validation:

```
$ curl -s --data-binary @codecov.yml https://codecov.io/validate
Valid!
```

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+coverage:
+  status:
+    project:
+      default:
+        target: auto # baseline is always the current v2.6.0 value
+        threshold: 1% # allow up to 1% drop before failing the check
+    patch:
+      default:
+        target: 80% # new/changed lines in a PR must be at least 80% covered
```
Comment (Luisfc68) [inline codecov.yml:14]: Since this is a smart contract project, I think there the 100% is not negotiable

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        threshold: 5% # allow up to 5% below target before failing
+
+comment:
+  layout: "reach,diff,flags,files"
+  behavior: default
+  require_changes: true # only post a comment when coverage actually changes
+
+ignore:
+  - "test/**" # forge test suites
+  - "script/**" # deployment and task scripts
```
Comment (Luisfc68) [inline codecov.yml]: these should be tested as well, in many cases they're critical

File: codecov.yml (hunk 1 above)
Comment (Freshenext) [reply codecov.yml:14]: 80% for now, we'll get to the point of having 100% incrementally.

File: codecov.yml (hunk 2 above)
Comment (Freshenext) [reply codecov.yml]: Added but with a note:

Both have no coverage so the report will show 40% aprox. We should follow up on this when we finish the PoC.

**Review by Luisfc68 — APPROVED** [review]
